doc/api/async_hooks.md

```diff
@@ -712,6 +712,69 @@ never be called.
 * Returns: {number} The same `triggerAsyncId` that is passed to the
 `AsyncResource` constructor.
 
+## Class: AsyncContext
+
+This class is used to create contexts that can be used through the event loop.
+
+### Example
+
+Let's build a logger that will always know the current HTTP request and use
+it to display enhanced logs without needing to explicitly pass the current
+HTTP request to it.
+
+```js
+const { AsyncContext } = require('async_hooks');
+const http = require('http');
+
+const kReq = 'CURRENT_REQUEST';
+const asyncContext = new AsyncContext();
+
+function log(...args) {
+  const store = asyncContext.getStore();
+  // Let's make sure the store exists and it contains a request
+  if (store && store.has(kReq)) {
+    const req = store.get(kReq);
+    // For instance, prints `GET /items ERR could not do something
+    console.log(req.method, req.url, ...args);
+  } else {
+    console.log(...args);
+  }
+}
+
+http.createServer((request, response) => {
+  asyncContext.enter((store) => {
+      store.set(kReq, request);
+      // some code
+       someAsyncOperation((err, result) => {
+        if (err){
+          log('ERR', err.message);
+          // ... rest of the code
+        }
+      });
+  });
+})
+.listen(8080);
+```
+
+### new AsyncContext()
+
+Creates a new instance of AsyncContext. Until the `enter` method is called, it
+does not provide any storage features.
+
+### asyncContext.enter(callback)
+
+Calling `asyncContext.enter(callback)` will create a new asynchronous context.
+This method will call the callback with a `Map` known as the store as argument.
+
```

Comment:
It should be clarified what happens when you create and enter a different `AsyncContext` from within an existing one... and that should be added to the test.

Reply:
Will do!

Reply:
Finally, done it!